tests/test_datasets/test_dataset_functions.py

```diff
@@ -1672,3 +1674,130 @@ def test_valid_attribute_validations(default_target_attribute, row_id_attribute,
         original_data_url=original_data_url,
         paper_url=paper_url,
     )
+
+    def test_delete_dataset(self):
+        data = [
+            ["a", "sunny", 85.0, 85.0, "FALSE", "no"],
+            ["b", "sunny", 80.0, 90.0, "TRUE", "no"],
+            ["c", "overcast", 83.0, 86.0, "FALSE", "yes"],
+            ["d", "rainy", 70.0, 96.0, "FALSE", "yes"],
+            ["e", "rainy", 68.0, 80.0, "FALSE", "yes"],
+        ]
+        column_names = ["rnd_str", "outlook", "temperature", "humidity", "windy", "play"]
+        df = pd.DataFrame(data, columns=column_names)
+        # enforce the type of each column
+        df["outlook"] = df["outlook"].astype("category")
+        df["windy"] = df["windy"].astype("bool")
+        df["play"] = df["play"].astype("category")
+        # meta-information
+        name = "%s-pandas_testing_dataset" % self._get_sentinel()
+        description = "Synthetic dataset created from a Pandas DataFrame"
+        creator = "OpenML tester"
+        collection_date = "01-01-2018"
+        language = "English"
+        licence = "MIT"
+        citation = "None"
+        original_data_url = "http://openml.github.io/openml-python"
+        paper_url = "http://openml.github.io/openml-python"
+        dataset = openml.datasets.functions.create_dataset(
+            name=name,
+            description=description,
+            creator=creator,
+            contributor=None,
+            collection_date=collection_date,
+            language=language,
+            licence=licence,
+            default_target_attribute="play",
+            row_id_attribute=None,
+            ignore_attribute=None,
+            citation=citation,
+            attributes="auto",
+            data=df,
+            version_label="test",
+            original_data_url=original_data_url,
+            paper_url=paper_url,
+        )
+        dataset.publish()
+        _dataset_id = dataset.id
+        self.assertTrue(openml.datasets.delete_dataset(_dataset_id))
+
+
+@mock.patch.object(requests.Session, "delete")
+def test_delete_dataset_not_owned(mock_get, test_files_directory):
+    openml.config.start_using_configuration_for_example()
+    content_file = (
+        test_files_directory / "mock_responses" / "datasets" / "data_delete_not_owned.xml"
+    )
+    mock_get.return_value = create_request_response(status_code=353, content_filepath=content_file)
```

Comment:
From my understanding of the docs, this should be HTTP error code `412`, and the OpenML error code `353` should only be used internally. If this correct, this needs to be changed for every function.

Reply:
You are right. I updated the tests. The reason the tests pass because `openml-python` does not look at the status code except that it is `!= 200`.

Reply:
But otherwise you're good with this testing approach?

Reply:
In general yes. I just don't know how we can check for such deviations from the API :( Maybe at some point we need to add a test suite that checks that the answers from the server are still the same XML that we expect :D